x-pack/agent/pkg/agent/program/program.go

```diff
@@ -90,3 +111,146 @@ func KnownProgramNames() []string {
 	}
 	return names
 }
+
+func groupBy(single *transpiler.AST) (map[string]*transpiler.AST, error) {
```

Comment:
minor thing, can you rename it to groupByOutput or something similar. i would expect additional argument describing grouping with `groupBy`

Reply:
Make sense, I think I was forward thinking too much here.